src/protocols/two_party_ecdsa/lindell_2017/party_one.rs

```diff
@@ -59,7 +59,7 @@ impl KeyGenFirstMsg {
             ec_context,
             &BigInt::sample_below(&EC::get_q().div_floor(&BigInt::from(3))),
         );
-        assert!(pk.mul_assign(ec_context, &sk).is_ok());
+        pk.mul_assign(ec_context, &sk).expect("Assignment expected");
```

Comment:
It seems like we can benefit from a cryptographic helper function that would call `pk.mul_assign(ec_context, &sk).expect("Failed to multiply and assign")` - so the protocol could use it without adding an `.expect(...)` at each call-site.
IIUC, the only way `mul_assign` can fail is if the secret key is [invalid](https://github.com/rust-bitcoin/rust-secp256k1/blob/ad080da1d815dfc170227c0c30de414180d29f95/src/key.rs#L116).

Reply:
https://github.com/KZen-networks/cryptography-utils/issues/8